addon/record-data.js

```diff
@@ -692,7 +692,10 @@ export default class FragmentRecordData extends RecordData {
       if (this._fragments[key]) {
         continue;
       }
-      if ((updates[key] === null) !== (original[key] === null)) {
+      if (
+        (updates[key] === null) !== (original[key] === null) ||
+        updates[key] !== original[key]
```

Comment:
This is a very weird conditional, could this be split into some variables that describe what is going on?

Reply:
I agree about the lack of readability of this. I made a truth table to help me understand the original:

```
|                                               | updates[key]===null | original[key]===null | Result |
| --------------------------------------------- | ------------------- | -------------------- | ------ |
| { updates[key] = "12", original[key] = null } | FALSE               | TRUE                 | TRUE   |
| { updates[key] = "12", original[key] = "ab" } | FALSE               | FALSE                | FALSE  |
| { updates[key] = null, original[key] = null } | TRUE                | TRUE                 | FALSE  |
| { updates[key] = null, original[key] = "ab" } | TRUE                | FALSE                | TRUE   |
```

It made me think that the second case (where the values are different) should return `true`, but after playing around with it I didn't need to make a change, just call `notify`.